core/modules/hubot/robot.js

```diff
@@ -0,0 +1,223 @@
+        if (cleanedLine.length === 0 || cleanedLine.toLowerCase() === 'none') {
+            continue;
+        }
+
+        var nextSection = cleanedLine.toLowerCase().replace(':', '');
+        if (hubotDocumentationSections.indexOf(nextSection) >= 0) {
+            currentSection = nextSection;
+            scriptDocumentation[currentSection] = [];
+        } else if (currentSection) {
+            scriptDocumentation[currentSection].push(cleanedLine.trim());
+            if (currentSection === 'commands') {
+                commands.push(cleanedLine.trim());
+            }
+        }
+    }
+
+    var help = [];
+    if (scriptDocumentation.commands) {
+        for (var i = 0; i < scriptDocumentation.commands.length; i++) {
+            var spl = scriptDocumentation.commands[i].match(/(?:[^-]|(?:--[^ ]))+/g);
+            if (spl[0].startsWith('hubot ')) {
+                spl[0] = '{{commandPrefix}}' + spl[0].substr(6);
+            }
+            for (var j = 0; j < spl.length; j++) {
+                spl[j] = spl[j].trim();
+            }
+            if (spl.length === 1) {
+                spl.push('Does what the command says.');
+            }
+            help.push(spl);
+        }
+    }
+
+    if (help.length === 0) {
+        help.push([scriptDocumentation.name, 'Does something. The unhelpful author didn\'t specify what.']);
+    }
+
+    return {
+        name: scriptDocumentation.name,
+        startup: scriptLocation,
+        version: 1.0,
+        dependencies: scriptDocumentation.dependencies,
+        configuration: scriptDocumentation.configuration,
+        notes: scriptDocumentation.notes,
+        authors: scriptDocumentation.authors || scriptDocumentation.author,
+        examples: scriptDocumentation.examples,
+        tags: scriptDocumentation.tags,
+        urls: scriptDocumentation.urls,
+        help: help
+    };
+};
+
+Robot.prototype.run = function (api, event) {
+    if (!event.__robotCallbackListeners) {
+        return;
+    }
+
+    for (var i = 0; i < event.__robotCallbackListeners.length; i++) {
+        var responder = new Responder(api, event, event.__robotCallbackListeners[i].match, event.__robotCallbackMessage);
+        event.__robotCallbackListeners[i].callback(responder);
+    }
+};
+
+Robot.prototype.match = function (event, commandPrefix) {
+    var msg = new Message(event, commandPrefix);
+
+    var hasMatched = false;
+    for (var i = 0; i < this.listeners.length; i++) {
+        var m = this.listeners[i].matcher(msg);
+        if (m) {
+            hasMatched = true;
+            if (!event.__robotCallbackListeners) {
+                event.__robotCallbackListeners = [];
+                event.__robotCallbackMessage = msg;
+            }
+            event.__robotCallbackListeners.push({
+                callback: this.listeners[i].callback,
+                match: m
+            });
+        }
+    }
+
+    if (!hasMatched) {
+        for (var i = 0; i < this.catchAllListeners.length; i++) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'i' is already declared in line 121.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530333298/issues/source?bid=3282987&fileBranchId=3282997#l137)